Link and Manfiest Verification CI-CD Checks, use FreeRTOS/CI-CD Formatting Action

Skptak opened the pull request:
<!--- Title -->

Description
-----------
Use ubuntu 20.04 for formatting so that this repo can use the same formatting as all other FreeRTOS Repos

CMSIS Is not directly included in this repo, removing it from this repo's manifest.yml file

Fix a broken link in the aws-iot-example.md file, remove the trailing comma from a link, and add an exclude for a bash link

Add the link and manifest verification workflow to this repo. Bump checkout actions to v3 to remove the deprecation warnings

Test Steps
-----------
<!-- Describe the steps to reproduce. -->

Checklist:
----------
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
<!-- If any, please provide issue ID. -->

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

urutva commented:
@Skptak Thanks for the changes. Can you please add `Signed-off-by: Full Name <email>` to each commit. We've an internal CI check to verify that every commit contains `Signed-off-by: Full Name <email>`. Please check [here](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) for more information. Thanks.

Skptak commented:
@urutva I fixed the commit message issue you mentioned, is that the change you needed/were looking for?
I squashed my commits while doing it, so now there's just the singular one as well.
